library/ssl_tls12_server.c

```diff
@@ -2531,7 +2531,7 @@ static int ssl_write_certificate_request( mbedtls_ssl_context *ssl )
         if( ! mbedtls_ssl_sig_alg_is_supported( ssl, *sig_alg ) )
             continue;
 
-        MBEDTLS_PUT_UINT16_BE( *sig_alg, p, sa_len );
+        MBEDTLS_PUT_UINT16_BE( *sig_alg, p, sa_len + 2 );
         sa_len += 2;
```

Comment:
Ok, incorporated this and because I initially thought this suggestion was also wrong (would write the wrong length), added comments to try and make this code more clear.